tests/core/test_mnemonic.py

```diff
@@ -0,0 +1,502 @@
+# Originally from: https://github.com/trezor/python-mnemonic
+#
+# Copyright (c) 2013 Pavol Rusnak
+# Copyright (c) 2017 mruddy
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+import pytest
+
+from eth_account.hdaccount.mnemonic import (
+    ConfigurationError,
+    Mnemonic,
+    normalize_string,
+)
+
+
+def test_failed_checksum():
+    mnemo = Mnemonic("english")
+    assert not mnemo.check(
+        "bless cloud wheel regular tiny venue bird web grief security dignity zoo"
+    )
+
+
+def test_detection():
+    assert "english" == Mnemonic.detect_language("security")
```

Comment:
For completeness' sake, what do you think about checking other languages as well?